libraries/ui-library/src/components/six-range/six-range.tsx

```diff
@@ -184,68 +179,109 @@ export class SixRange {
     this.invalid = false;
   }
 
-  handleInput() {
-    this.value = Number(this.input.value);
-    this.sixChange.emit();
-
-    requestAnimationFrame(() => this.syncTooltip());
-    requestAnimationFrame(() => this.calculateColorRunnableTrack());
-  }
+  private handleInput = () => {
+    if (this.nativeInput != null) {
+      this.update(parseFloat(this.nativeInput.value));
+    }
+    requestAnimationFrame(() => {
+      this.sixChange.emit();
+    });
+  };
 
-  handleBlur() {
+  private handleBlur = () => {
     this.hasFocus = false;
     this.hasTooltip = false;
     this.sixBlur.emit();
-    this.resizeObserver.unobserve(this.input);
-  }
+    if (this.nativeInput != null) {
+      this.resizeObserver?.unobserve(this.nativeInput);
+    }
+  };
 
-  handleFocus() {
+  private handleFocus = () => {
     this.hasFocus = true;
     this.hasTooltip = true;
     this.sixFocus.emit();
-    this.resizeObserver.observe(this.input);
-  }
+    if (this.nativeInput != null) {
+      this.resizeObserver?.observe(this.nativeInput);
+    }
+  };
 
-  handleSlotChange() {
+  private handleSlotChange = () => {
     this.hasHelpTextSlot = hasSlot(this.host, 'help-text');
     this.hasErrorTextSlot = hasSlot(this.host, 'error-text');
     this.hasLabelSlot = hasSlot(this.host, 'label');
-  }
+  };
 
-  handleTouchStart() {
+  private handleTouchStart = () => {
     this.setFocus();
-  }
+  };
 
-  displayError() {
+  private displayError() {
     return this.invalid && (!this.errorOnBlur || !this.hasFocus);
   }
 
-  syncTooltip() {
-    if (this.tooltip !== 'none') {
-      const percent = Math.max(0, (this.value - this.min) / (this.max - this.min));
-      const inputWidth = this.input.offsetWidth;
+  private syncTooltip(min: number, max: number, value: number) {
+    if (this.tooltip !== 'none' && this.nativeInput != null && this.output != null) {
+      const percent = Math.max(0, (value - min) / (max - min));
+      const inputWidth = this.nativeInput.offsetWidth;
       const tooltipWidth = this.output.offsetWidth;
-      const thumbSize = getComputedStyle(this.input).getPropertyValue('--thumb-size');
+      const thumbSize = getComputedStyle(this.nativeInput).getPropertyValue('--thumb-size');
       const x = `calc(${inputWidth * percent}px - calc(calc(${percent} * ${thumbSize}) - calc(${thumbSize} / 2)))`;
       this.output.style.transform = `translateX(${x})`;
       this.output.style.marginLeft = `-${tooltipWidth / 2}px`;
     }
   }
 
+  private isFirefox() {
+    return navigator.userAgent.toLowerCase().indexOf('firefox') > -1;
+  }
+
+  private update(updateValue?: number) {
+    if (updateValue != null) {
+      this.value = updateValue;
+    }
+
+    const min = Number(this.min) ?? 0;
+    const max = Number(this.max) ?? 0;
+    let value;
+    const parsedValue = parseFloat(this.value as never);
+    if (isNaN(parsedValue)) {
+      value = this.getDefaultValue(min, max);
+    } else {
+      value = parsedValue;
+    }
```

Comment:
Could extract a small function to get rid of the `let`:

const getValue = (min: number, max: number) => {
    const parsedValue = parseFloat(this.value as never);
    if (isNaN(parsedValue)) {
        return this.getDefaultValue(min, max);
    } else {
        return parsedValue;
    }
}

...
const value = getValue(min, max);

Reply:
I'm unsure if it's worth it. I guess the goal would be to prevent unintentional value reassignment. For such a small function, I prefer simplicity, but it's, of course, a matter of taste.